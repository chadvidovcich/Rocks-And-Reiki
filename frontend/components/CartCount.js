import { CSSTransition, TransitionGroup } from 'react-transition-group';
import styled from 'styled-components';

const Dot = styled.div`
  background: var(--red);
  color: white;
  border-radius: 50%;
  padding: 0.5rem;
  line-height: 2rem;
  margin-left: 1rem;
  font-feature-settings: 'tnum';
  font-variant-numeric: tabular-nums;
`;

function CartCount({ count }) {
  return (
    <TransitionGroup>
      <CSSTransition
        unmountOnExit
        classNames={count}
        className="count"
        key={count}
        timeout={{ enter: 5000, exit: 5000 }}
      >
        <Dot>{count}</Dot>
      </CSSTransition>
    </TransitionGroup>
  );
}
export default CartCount;
