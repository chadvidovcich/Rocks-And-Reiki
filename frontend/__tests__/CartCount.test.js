// @ts-ignore
import { render, screen } from '@testing-library/react';
// @ts-ignore
import wait from 'waait';
import CartCount from '../components/CartCount';

describe('<CartCount/>', () => {
  it('renders', () => {
    render(<CartCount count={10} />);
  });

  it('matches snapshot', () => {
    const { container } = render(<CartCount count={31} />);
    expect(container).toMatchSnapshot();
  });

  it('updates via props', async () => {
    const { container, rerender, debug } = render(<CartCount count={31} />);
    expect(container).toHaveTextContent('31');

    rerender(<CartCount count={'32'} />);
    // wait for 400ms
    expect(container).toHaveTextContent('3231');
    await wait(400);
    expect(container).toHaveTextContent('32');
    expect(container).toMatchSnapshot();
  });
});
