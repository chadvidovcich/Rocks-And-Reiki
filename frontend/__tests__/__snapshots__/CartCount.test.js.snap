// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount/> Matches snapshot 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-1waysp3-1 kgJmyN"
  >
    <div>
      <div
        class="CartCount__Dot-sc-1waysp3-0 cXpbJa count"
      >
        31
      </div>
    </div>
  </span>
</div>
`;

exports[`<CartCount/> Updates via props 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-1waysp3-1 kgJmyN"
  >
    <div>
      <div
        class="CartCount__Dot-sc-1waysp3-0 cXpbJa count count-enter-done"
      >
        32
      </div>
    </div>
  </span>
</div>
`;

exports[`<CartCount/> matches snapshot 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-1waysp3-1 kgJmyN"
  >
    <div>
      <div
        class="CartCount__Dot-sc-1waysp3-0 cXpbJa count"
      >
        31
      </div>
    </div>
  </span>
</div>
`;

exports[`<CartCount/> updates via props 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-1waysp3-1 kgJmyN"
  >
    <div>
      <div
        class="CartCount__Dot-sc-1waysp3-0 cXpbJa count count-enter-done"
      >
        32
      </div>
    </div>
  </span>
</div>
`;
